src/property-filter/utils.ts

```diff
@@ -69,7 +69,10 @@ export function matchTokenValue(token: Token, filteringOptions: readonly Interna
       // exact match found: return it
       return { ...token, value: option.value };
     }
-    if (token.value.toLowerCase() === (option.label ?? option.value ?? '').toLowerCase()) {
+    if (
+      typeof token.value === 'string' &&
```

Comment:
If we have to make this type check, doesn't that mean that our Typescript types are inaccurate? Can we fix that?

Reply:
There is an ESLint rule that can detect this.

It currently shows:
25 reports of such “[unsafe calls](https://typescript-eslint.io/rules/no-unsafe-call/)”
61 reports of “[unsafe assignments](https://typescript-eslint.io/rules/no-unsafe-assignment/)” to variables
43 reports of “[unsafe arguments](https://typescript-eslint.io/rules/no-unsafe-argument/)” to functions
and 14 reports of “[unsafe returns](https://typescript-eslint.io/rules/no-unsafe-return)” in functions

We have this task as a internal BRF ticket.

Reply:
So... looking at the type definition for the `token.value`, coming from the [collection hooks package](https://github.com/cloudscape-design/collection-hooks/blob/main/src/interfaces.ts#L144), it has to be `any`: by default, the token value is a `string` - however, when a custom property is used, the token value can be `any`.

I added a comment here to also make it more clear.

Reply:
Got it, thank you!